### ISSUE TITLE
openstack_compute_instance does not support state SHELVE_OFFLOADED

### ISSUE THREAD

**Issue by qhua948** [terraform-providers/terraform-provider-openstack#941]
Hi there,

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see https://www.terraform.io/community.html.

### Terraform Version
`Terraform v0.12.16`

### Affected Resource(s)
- openstack_compute_instance_v2

### Terraform Configuration Files
```tf
resource "openstack_compute_instance_v2" "app" {
  name            = "${var.name}-app"
  key_pair        = var.key_pair
  flavor_id       = data.openstack_compute_flavor_v2.c1_c8r8.id
  user_data       = data.template_cloudinit_config.config.rendered
  security_groups = var.security_groups
  block_device {
    uuid                  = data.openstack_images_image_v2.default.id
    source_type           = "image"
    volume_size           = 10
    destination_type      = "volume"
    delete_on_termination = true
  }
  network {
    name = "front-end"
  }
}

```

### Debug Output
https://gist.github.com/qhua948/a3ecc37f77b6b8a20ba41e6fc5e250e0

### Expected Behavior
It should accept `SHELVE_OFFLOADED` as a `power_state`

### Actual Behavior
openstack provider throws error on `resource_openstack_compute_instance_v2.go:668`
https://github.com/terraform-providers/terraform-provider-openstack/blob/73adf632a9bc644d61c39398356e0c58131ce3e3/openstack/resource_openstack_compute_instance_v2.go

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. Has an instance managed by the provider and set the state to SHELVED
2. `terraform plan` with the configuration

**Comment by qhua948** [comment]
I believe this has something to do with the upstream gophercloud where shelving/unshelving is not supporting?

**Comment by qhua948** [comment]
#942

**Comment by ozerovandrei** [comment]
Fixed in #942
